tests/test_execution_compute_plan.py

```diff
@@ -379,6 +379,68 @@ def test_compute_plan_aggregate_composite_traintuples(global_execution_env):
         assert t.status == assets.Status.done
 
 
+@pytest.mark.slow
+def test_compute_plan_remove_intermediary_model(global_execution_env):
+    """
+    Create a simple compute plan with clean_models at true, see it done and
+    create a test tuple on a intermediary model. Expect it to fail at the
+    execution.
+    """
+
+    factory, initial_assets, network = global_execution_env
+    client = network.clients[0]
+
+    # register objectives, datasets, and data samples
+    initial_assets = initial_assets.filter_by(client.node_id)
+    dataset = initial_assets.datasets[0]
+    data_sample_1, data_sample_2, data_sample_3, _ = dataset.train_data_sample_keys
+    objective = initial_assets.objectives[0]
+
+    # register algo
+    spec = factory.create_algo()
+    algo = client.add_algo(spec)
+
+    # create a compute plan with clean_model activate
+    cp_spec = factory.create_compute_plan(clean_models=True)
+
+    traintuple_spec_1 = cp_spec.add_traintuple(
+        algo=algo,
+        dataset=dataset,
+        data_samples=[data_sample_1]
+    )
+    cp_spec.add_testtuple(
+        objective=objective,
+        traintuple_spec=traintuple_spec_1,
+    )
+
+    traintuple_spec_2 = cp_spec.add_traintuple(
+        algo=algo,
+        dataset=dataset,
+        data_samples=[data_sample_2],
+        in_models=[traintuple_spec_1]
+    )
+    cp_spec.add_testtuple(
+        objective=objective,
+        traintuple_spec=traintuple_spec_2,
+    )
+
+    cp = client.add_compute_plan(cp_spec).future().wait()
+
+    # All the train/test tuples should succeed
+    for t in cp.list_traintuple() + cp.list_testtuple():
+        assert t.status == assets.Status.done
```

Comment:
This is not necessary as it is already checked in the wait() method.

Reply:
Doesn't it just test that it finished but not done?

Reply:
`wait()` raises by default an exception if the status is canceled or failed. It can be changed if setting the input argument `raises` to `False`.